x/programs/examples/imports/balance/balance.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package balance
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"github.com/bytecodealliance/wasmtime-go/v12"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/x/programs/examples/storage"
+	"github.com/ava-labs/hypersdk/x/programs/runtime"
+)
+
+const Name = "balance"
+
+func New(log logging.Logger, mu state.Mutable) *Import {
+	return &Import{mu: mu, log: log}
+}
+
+type Import struct {
+	mu    state.Mutable
+	log   logging.Logger
+	meter runtime.Meter
+}
+
+func (i *Import) Name() string {
+	return Name
+}
+
+func (i *Import) Register(link runtime.Link, meter runtime.Meter) error {
+	if err := link.FuncWrap(Name, "get", i.getFn); err != nil {
+		return err
+	}
+	if err := link.FuncWrap(Name, "set", i.setFn); err != nil {
+		return err
+	}
+	if err := link.FuncWrap(Name, "add", i.addFn); err != nil {
+		return err
+	}
+	if err := link.FuncWrap(Name, "subtract", i.subFn); err != nil {
+		return err
+	}
+	if err := link.FuncWrap(Name, "delete", i.delFn); err != nil {
+		return err
+	}
+
+	i.meter = meter
+
+	return nil
+}
+
+func (i *Import) getFn(caller *wasmtime.Caller, assetPtr, keyPtr int32) int32 {
+	pk, err := runtime.PublicKeyFromOffset(caller, storage.HRP, keyPtr)
+	if err != nil {
+		i.log.Error("failed to read key from memory",
+			zap.Error(err),
+		)
+		return -1
+	}
+	assetID, err := runtime.IDFromOffset(caller, assetPtr)
+	if err != nil {
+		i.log.Error("failed to read assetID from memory",
+			zap.Error(err),
+		)
+		return -1
+	}
+
+	// TODO: add cost for state access?
+	// i.meter.Spend(BalanceCost)
+
+	balance, err := storage.GetBalance(context.Background(), i.mu, pk, assetID)
+	if err != nil {
+		i.log.Error("failed to get balance",
+			zap.Error(err),
+		)
+		return -1
+	}
+
+	return int32(balance)
+}
+
+func (i *Import) setFn(caller *wasmtime.Caller, assetPtr, keyPtr, balance int32) int32 {
+	pk, err := runtime.PublicKeyFromOffset(caller, storage.HRP, keyPtr)
+	if err != nil {
+		i.log.Error("failed to read key from memory",
+			zap.Error(err),
+		)
+		return -1
+	}
+	assetID, err := runtime.IDFromOffset(caller, assetPtr)
+	if err != nil {
+		i.log.Error("failed to read assetID from memory",
+			zap.Error(err),
+		)
+		return -1
+	}
+
+	// TODO: add cost for state access?
+	// i.meter.Spend(BalanceCost)
```

Comment:
This will be handled by the HyperSDK outside of this context. All we need is for this to capture compute cost.

You should leave a comment indicating as much.

Reply:
We should still charge for any "context switch" operation but it shouldn't be based on state. I think we could start with a single `ContextSwitchCost` in `cfg` and use that for all of these?